core/metacling/src/TCling.cxx

```diff
@@ -1425,6 +1425,10 @@ TCling::TCling(const char *name, const char *title, const char* const argv[])
       clingArgsStorage.push_back("-Wno-undefined-inline");
       clingArgsStorage.push_back("-fsigned-char");
       clingArgsStorage.push_back("-O1");
+      // Disable optimized register allocation which is turned on automatically
+      // by -O1, but seems to require -O2 to not explode in run time.
+      clingArgsStorage.push_back("-mllvm");
+      clingArgsStorage.push_back("-optimize-regalloc=0");
```

Comment:
Do we know of a *negative* perf impact?

Reply:
In principle, the generated code can be slower, I didn't really test because I focused on JIT time. However, it can't be worse than `-O0` I would say, and we still have all the IR optimizations enabled.

Reply:
Is that the reason `stressGraphics` blew up when we enabled `-O1`?

Reply:
No, `stressGraphics` blew up because of the very long functions, see also #9312. For RDF, we're now avoiding this situation thanks to Enrico's https://github.com/root-project/root/pull/9328.